ssoPlugin/widget/index.html

```diff
@@ -0,0 +1,278 @@
+			padding: 0;
+		}
+		.bif-sso__alert-message--success {
+			color: #00b500;
+		}
+		.bif-sso__alert-message--error {
+			color: #ec0000;
+		}
+	</style>
+</head>
+<body class="uninitialized">
+
+	<div id="bif-sso__banner">Widget - SSO</div>
+
+	<div id="bif-sso__wrapper">
+		<div id="bif-sso__alert-message"></div>
+
+		<div id="bif-sso__content--uninitialized" class="bif-sso__content">
+			<p>Initializing...</p>
+		</div>
+
+		<div id="bif-sso__content--unauthenticated" class="bif-sso__content">
+			<p>Click the button below to authenticate.</p>
+			<button id="bif-sso__login-button">Log In</button>
+		</div>
+
+		<div id="bif-sso__content--authenticated" class="bif-sso__content">
+			<p>Successfully authenticated!</p>
+		</div>
+	</div>
+
+	<script type="text/javascript">
+		// ============================================================================================
+		// UTILITY METHODS
+		// ============================================================================================
+
+		/**
+		 * Runs the provided function when the document is fully loaded.
+		 *
+		 * @param {Function} fn The function to be run.
+		 */
+		const ready = (fn) => {
+			if (document.readyState != 'loading') fn()
+			else document.addEventListener('DOMContentLoaded', fn)
+		}
+
+		/**
+		 * Saves the provided data object to the browser's localStorage using the BuildFire library.
+		 *
+		 * @param   {Object}  data An object containing the data to save.
+		 * @returns {Promise}      Resolves with boolean true, rejects with an error message.
+		 */
+		const saveToLocalStorage = (data) => new Promise((resolve, reject) => {
+			buildfire.localStorage.setItem('@bif/sso', data, (err, result) => {
+				if (err || !result) reject('There was a problem saving to local storage.')
+				else resolve(true)
+			})
+		})
+
+		// ============================================================================================
+		// APPLICATION METHODS
+		// ============================================================================================
+
+		// Set DOM variables that we'll reference in the following methods thanks to lexical scoping.
+		const $body = document.querySelector('body')
+		const $alertMessage = document.getElementById('bif-sso__alert-message')
+		const $loginButton = document.getElementById('bif-sso__login-button')
+
+		/**
+		 * Marks the completion of initialization by changing some body classes.
+		 */
+		const initComplete = () => {
+			$body.classList.remove('uninitialized')
+			$body.classList.add('initialized')
+		}
+
+		/**
+		 * Updates the alert message placeholder with the provided message and type.
+		 *
+		 * Expected types are "error" or "success". Default is "notification" and has no special style.
+		 *
+		 * @param {String}  message      The message that should be displayed.
+		 * @param {String} messageType? The message type, from the list above.
+		 */
+		const alertMessage = (message, messageType = 'notification') => {
```

Comment:
Since this method is repeated, should we have this be in a common utils.js somewhere?

Reply:
Yes, but if it's okay I'd like to reserve that for #8 so I can explore the best way. (Assets can't natively be shared between the control and widget portion of the codebase. BuildFire does support Webpacked plugins but we'd need to convert this over to that style, which I'm down to do but hesitant to spend cycles until we've got the other big TODOs complete.)